### ISSUE TITLE
Should not set readable if statusCode is 204

### ISSUE THREAD

**Issue by mikicho** [mswjs/interceptors#433]
https://github.com/mswjs/interceptors/blob/main/src/interceptors/ClientRequest/utils/createResponse.ts#L21

![image](https://github.com/mswjs/interceptors/assets/11459632/f3943f9b-0714-4c9d-a204-9fee06f7f713)

**Comment by mikicho** [comment]
Also for 101, 103, 204, 205, or 304.
https://fetch.spec.whatwg.org/#statuses

**Comment by kettanaito** [comment]
I've opened a fix for this in #436. @mikicho, note that while status codes 101 and 103 also forbid response from having a body, you cannot even construct responses with those status codes. The `ResponseInit.status` expects a [200, 599] status code range.

```js
new Response(null, { status: 101 })
// Uncaught RangeError: Failed to construct 'Response': The status provided (101) is outside the range [200, 599].
    at <anonymous>:1:1
```

The library won't account for those status codes because it's unnecessary.

**Comment by kettanaito** [comment]
## Released: v0.25.3 🎉

This has been released in v0.25.3!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.25.3)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.25.3)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
